### PR TITLE
Bump github.com/golangci/golangci-lint to to 1.50.1 and apply fixes

### PR DESCRIPTION
**Pull request by pellared**
Supersedes https://github.com/signalfx/splunk-otel-go/pull/1553

- Bump golangci-lint
- Update golangci-lint configuration
- Use `golangci-lint --fix` instead of `gofumpt` + `goimports`
- Apply fixes for lint errors
- Update the build pipeline to use Go 1.19

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -114,7 +114,6 @@ linters:
   disable-all: true
   enable:
     - bodyclose
-    - deadcode
```
Comment (pellared) [inline .golangci.yml:117]: deprecated

File: .golangci.yml [hunk 2]
```diff
@@ -87,10 +87,10 @@ linters-settings:
     ignore-words:
       - importas
   nolintlint:
-    allow-leading-space: true # don't require machine-readable nolint directives (i.e. with no leading space)
+    allow-leading-space: false # don't require machine-readable nolint directives (i.e. with no leading space)
     allow-unused: false # report any unused nolint directives
-    require-explanation: false # don't require an explanation for nolint directives
-    require-specific: false # don't require nolint directives to be specific about which linter is being skipped
+    require-explanation: true # don't require an explanation for nolint directives
+    require-specific: true # don't require nolint directives to be specific about which linter is being skipped
```
Comment (pellared) [inline .golangci.yml:93]: stronger requirements applied

File: .golangci.yml [hunk 3]
```diff
@@ -137,15 +136,13 @@ linters:
     - nakedret
     - noctx
     - nolintlint
-    - rowserrcheck
+    # - rowserrcheck # rowserrcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
```
Comment (pellared) [inline .golangci.yml:139]: to remove warnings from output

File: .golangci.yml (hunk 2 above)
Comment (pellared) [reply .golangci.yml]: I fixed the comments here https://github.com/signalfx/splunk-otel-go/pull/1571/commits/45f17ed104a5d0f2ffdc4ccab88ca718869b3474

File: example/main.go [hunk 4]
```diff
@@ -12,6 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// This example instruments a simple HTTP server-client application.
```
Comment (MrAlias) [inline example/main.go:15]:
```suggestion
// Package example instruments a simple HTTP server-client application.
```

**Review by MrAlias — APPROVED** [review]

File: example/main.go (hunk 4 above)
Comment (pellared) [reply example/main.go]: @MrAlias Are you sure? This is a main package

Comment (MrAlias) [reply example/main.go]: Yep

**Pull request merged**